js/modules/k6/ws/ws.go

```diff
@@ -123,6 +129,26 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 					continue
 				}
 				for _, key := range headersObj.Keys() {
+					if key == HeaderKeyCookie {
+						req, err := http.NewRequest("", url, nil) //nolint:noctx
```

Comment:
Why do you need to create a new request here or mess with the `Cookie` header? Couldn't you create `http.Cookie` directly and pass it to `jar.SetCookies()`?

Reply:
On a somewhat related note ... this is not how it works for HTTP requests. 

There is a [default jar](https://github.com/loadimpact/k6/blob/master/lib/state.go#L57) and you can also provide a CookieJar and there is some logic to merge between all of those and Cookie param, which probably should be reused here.

Reply:
> Why do you need to create a new request here or mess with the Cookie header? Couldn't you create http.Cookie directly and pass it to jar.SetCookies()?

Because parsing HTTP cookies is a private function inside go standard library ([proof](https://github.com/golang/go/blob/master/src/net/http/cookie.go#L238)) and I decided not to copy this function.